src/components/media/controls/controls.tsx

```diff
@@ -0,0 +1,292 @@
+/* eslint-disable no-magic-numbers */
+import './controls.css'
+
+import Forward10RoundedIcon from '@mui/icons-material/Forward10Rounded'
+import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown'
+import KeyboardArrowUpIcon from '@mui/icons-material/KeyboardArrowUp'
+import PauseCircleFilledIcon from '@mui/icons-material/PauseCircleFilled'
+import PlayCircleFilledIcon from '@mui/icons-material/PlayCircleFilled'
+import Replay10RoundedIcon from '@mui/icons-material/Replay10Rounded'
+import VolumeOffRoundedIcon from '@mui/icons-material/VolumeOffRounded'
+import VolumeUpRoundedIcon from '@mui/icons-material/VolumeUpRounded'
+import Box from '@mui/material/Box'
+import Button from '@mui/material/Button'
+import IconButton from '@mui/material/IconButton'
+import Slider from '@mui/material/Slider'
+import Typography from '@mui/material/Typography'
+import React, { useState } from 'react'
+
+import { formatDurationTime } from '../../helper'
+
+interface MediaIconButtonProps {
+  onClick: () => void
+  ariaLabel: string
+  icon: React.ReactNode
+  className?: string
+  dataCy?: string
+}
+
+export function MediaIconButton(props: MediaIconButtonProps) {
+  return (
+    <IconButton
+      onClick={props.onClick}
+      aria-label={props.ariaLabel}
+      className={props.className}
+      data-cy={props.dataCy}
+    >
+      {props.icon}
+    </IconButton>
+  )
+}
+
+interface MediaControls {
+  mediaElement: HTMLMediaElement
+}
+
+export function ProgressSlider(props: MediaControls) {
+  const formattedElapsedTime = formatDurationTime(
+    props.mediaElement.currentTime
+  )
+
+  function handleTimeChange(event: Event, newValue: number | number[]) {
+    props.mediaElement.currentTime = newValue as number
+  }
+
+  function renderBufferedProgressBar() {
+    const buffered = props.mediaElement.buffered
+    let totalBufferedDuration = 0
+
+    for (let i = 0; i < buffered.length; i++) {
+      totalBufferedDuration += buffered.end(i) - buffered.start(i)
+    }
+
+    const bufferedWidth =
+      (totalBufferedDuration / props.mediaElement.duration) * 100
+
+    return (
+      <Box
+        className="rustic-progress-buffered"
+        sx={{
+          backgroundColor: 'primary.light',
+          width: `${bufferedWidth}%`,
+        }}
+      />
```

Comment:
MUI LinearProgress component with determinate variant is a better fit for this.

References
https://mui.com/material-ui/react-progress/#linear-determinate